lib/rack/lint.rb

```diff
@@ -304,14 +317,12 @@ def check_environment(env)
         ## is reserved for use with the Rack core distribution and other
         ## accepted specifications and must not be used otherwise.
         ##
-        %w[REQUEST_METHOD SERVER_NAME QUERY_STRING SERVER_PROTOCOL rack.errors].each do |header|
-          raise LintError, "env missing required key #{header}" unless env.include? header
-        end
 
         ## The <tt>SERVER_PORT</tt> must be an Integer if set.
-        server_port = env["SERVER_PORT"]
-        unless server_port.nil? || (Integer(server_port) rescue false)
-          raise LintError, "env[SERVER_PORT] is not an Integer"
+        if server_port = env[SERVER_PORT]
+          unless (Integer(server_port) rescue false)
```

Comment:
As this was the previous code, I don't think you need to change it in this PR, but I don't think this check is correct.  SPEC says `must be an Integer`, not `must be convertible to an Integer`, so the correct check here is `server_port.is_a?(Integer)`.

Reply:
I intend to rebase and merge this PR (FF), so I'll add a fix for this as a separate commit.

Reply:
Hmm, okay, I tried this but "The <tt>SERVER_PORT</tt> must be an Integer if set." directly conflicts with all values must be strings?

Reply:
Interesting.  I would assume in order to pass Lint, servers are providing a string?  Puma appears to pass string, so I think we should change SPEC to state `SERVER_PORT` should be all digits as well (just like `CONTENT_LENGTH`).

Reply:
https://github.com/rack/rack/issues/2293